src/v0/destinations/intercom_v2/utils.js

```diff
@@ -99,10 +101,18 @@ const getResponse = (method, endpoint, headers, payload) => {
 
 const searchContact = async (event) => {
   const { message, destination, metadata } = event;
-  const lookupField = getLookUpField(message);
-  let lookupFieldValue = getFieldValueFromMessage(message, lookupField);
-  if (!lookupFieldValue) {
-    lookupFieldValue = message?.context?.traits?.[lookupField];
+  const messageType = getEventType(message);
+  let lookupField = null;
+  let lookupFieldValue = null;
+  if (messageType === EventType.RECORD) {
+    const { identifiers } = message;
+    [[lookupField, lookupFieldValue]] = Object.entries(identifiers);
+  } else {
+    lookupField = getLookUpField(message);
+    lookupFieldValue = getFieldValueFromMessage(message, lookupField);
+    if (!lookupFieldValue) {
+      lookupFieldValue = message?.context?.traits?.[lookupField];
+    }
```

Comment:
Just a suggestion to make it more readable:
You can probably encapsulate the if and else statement into a function inside itself for more readability and separation of message parsing which is not related to seachContact.

```
const { message, destination, metadata } = event;
  const messageType = getEventType(message);

  const extractFieldAndValue = (message, messageType) => {
    if (messageType === EventType.RECORD) {
      const { identifiers } = message;
      return Object.entries(identifiers)[0];
    }
    const lookupField = getLookUpField(message);
    let lookupFieldValue = getFieldValueFromMessage(message, lookupField) || 
                           message?.context?.traits?.[lookupField];
    return [lookupField, lookupFieldValue];
  };

  const [lookupField, lookupFieldValue] = extractFieldAndValue(message, messageType);

  if (!lookupField || !lookupFieldValue) {
    throw new Error('Missing lookup field or value for searchContact');
  }
```

Reply:
done.